src/ashley/models.py

```diff
@@ -64,3 +70,100 @@ class Meta:
 
 # Default implementation of the AbstractUser, using django-machina's model factory
 User = model_factory(AbstractUser)
+
+
+class AbstractLTIContext(Model):
+    """Abstract LTI Context model for Ashley."""
+
+    GROUP_PREFIX = "cg"
+    GROUP_DELIMITER = "__"
+
+    remote_id = models.CharField(max_length=150, null=False, blank=False, db_index=True)
+
+    lti_consumer = models.ForeignKey(
+        LTIConsumer, on_delete=models.CASCADE, null=False, db_index=True
+    )
+
+    def base_group(self) -> Group:
```

Comment:
For such method getters, I think you should give names that defer from property names. 
I propose naming them : `get_*` so here:
```suggestion
    def get_base_group(self) -> Group:
```